spyderlib/utils/introspection/plugin_manager.py

```diff
@@ -326,6 +326,24 @@ def show_object_info(self, position, auto=True):
         info = self._get_code_info('info', position, auto=auto)
         self._handle_request(info)
 
+    def rename(self, code, position, new_name):
+        """Returns code with identifier renamed.
+        This function tries all plugins and uses the first one which 
+        has an attribute 'rename'. If no plugin has this attribute then 
+        an NotImplementedException is raised.
+        code (str): old source code
+        position (int): position of identifier to be renamed
+        new_name (str): new name of identifier
+        """
+        for plugin in self.plugins.values():
+            try:
+                fun = plugin.rename
+            except AttributeError: 
+                continue
+            return fun(code, position, new_name)
+        else:
+            raise NotImplementedError

```

Comment:
It seems strange to check here if the renaming is possible. If there is no plugin to rename variables, the menu entry should be hidden or disabled. This would also avoid raising `NotImplementedError` in the internal console.